From stable-bounces@linux.kernel.org  Wed Mar 12 14:35:07 2008
From: Haavard Skinnemoen <hskinnemoen@atmel.com>
To: stable@kernel.org
Date: Mon,  3 Mar 2008 16:51:48 +0100
Message-Id: <1204559508-13480-1-git-send-email-hskinnemoen@atmel.com>
Cc: anemo@mba.ocn.ne.jp, hskinnemoen@atmel.com, jeff@garzik.org
Subject: macb: Fix speed setting

From: Atsushi Nemoto <anemo@mba.ocn.ne.jp>
commit: 179956f498bd8cc55fb803c4ee0cf18be59c8b01

Fix NCFGR.SPD setting on 10Mbps.  This bug was introduced by
conversion to generic PHY layer in kernel 2.6.23.

Signed-off-by: Atsushi Nemoto <anemo@mba.ocn.ne.jp>
Signed-off-by: Jeff Garzik <jeff@garzik.org>
Signed-off-by: Haavard Skinnemoen <hskinnemoen@atmel.com>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
---

 drivers/net/macb.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/macb.c
+++ b/drivers/net/macb.c
@@ -148,7 +148,7 @@ static void macb_handle_link_change(stru
 
 			if (phydev->duplex)
 				reg |= MACB_BIT(FD);
-			if (phydev->speed)
+			if (phydev->speed == SPEED_100)
 				reg |= MACB_BIT(SPD);
 
 			macb_writel(bp, NCFGR, reg);
